src/modules/Embed/Embed.js

```diff
@@ -0,0 +1,198 @@
+import Icon from '../../elements/Icon'
+
+const _meta = {
+  name: 'Embed',
+  type: META.TYPES.MODULE,
+  props: {
+    aspectRatio: ['4:3', '16:9', '21:9'],
+    source: ['youtube', 'vimeo'],
+  },
+}
+
+/**
+ * An embed displays content from other websites like YouTube videos or Google Maps.
+ */
+export default class Embed extends Component {
+  static autoControlledProps = [
+    'active',
+  ]
+
+  static defaultProps = {
+    icon: 'video play',
+  }
+
+  static _meta = _meta
+
+  static propTypes = {
+    /** An element type to render as (string or function). */
+    as: customPropTypes.as,
+
+    /** An embed can be active. */
+    active: PropTypes.bool,
+
+    /** An embed can specify an alternative aspect ratio. */
+    aspectRatio: PropTypes.oneOf(_meta.props.aspectRatio),
+
+    /** Primary content. */
+    children: PropTypes.node,
+
+    /** Additional classes. */
+    className: PropTypes.string,
+
+    /** Initial value of active. */
+    defaultActive: PropTypes.bool,
+
+    /** Specifies an icon to use with placeholder content. */
+    icon: customPropTypes.itemShorthand,
+
+    /** Сalled with event on Embed click with (event, props). */
+    onClick: PropTypes.func,
+
+    /** A placeholder image for embed. */
+    placeholder: PropTypes.string,
+
+    /** Specifies a source to use. */
+    source: customPropTypes.every([
+      customPropTypes.disallow(['sourceUrl']),
+      PropTypes.oneOf(_meta.props.source),
+    ]),
+
+    /** Setting to true or false will force autoplay. */
+    sourceAutoPlay: customPropTypes.every([
+      customPropTypes.demand(['source']),
+      PropTypes.bool,
+    ]),
+
+    /** Whether to show networks branded UI like title cards, or after video calls to action. */
+    sourceBrandedUI: customPropTypes.every([
+      customPropTypes.demand(['source']),
+      PropTypes.bool,
+    ]),
+
+    /** Specifies a default chrome color with Vimeo or YouTube. */
+    sourceColor: customPropTypes.every([
+      customPropTypes.demand(['source']),
+      PropTypes.string,
+    ]),
+
+    /** Whether to show networks branded UI like title cards, or after video calls to action. */
+    sourceHd: customPropTypes.every([
+      customPropTypes.demand(['source']),
+      PropTypes.bool,
+    ]),
+
+    /** Specifies an id for source. */
+    sourceId: customPropTypes.every([
+      customPropTypes.demand(['source']),
+      PropTypes.string,
+    ]),
+
+    /** Specifies a url to use for embed. */
+    sourceUrl: customPropTypes.every([
+      customPropTypes.disallow(['source']),
+      PropTypes.string,
+    ]),
+  }
+
+  state = {}
+
+  handleClick = (e) => {
+    const { onClick } = this.props
+    const { active } = this.state
+
+    if (onClick) onClick(e, this.props)
+    if (!active) this.trySetState({ active: true })
+  }
+
+  getSource() {
+    const {
+      source,
+      sourceAutoPlay = true,
+      sourceBrandedUI = false,
+      sourceColor = '#444444',
+      sourceHd = true,
```

Comment:
Just a note, I'm have there same problems with `demand` like in #827, so I left values there instead of `defaultProps`.